examples/go-client/main.go

```diff
@@ -18,26 +18,76 @@ package main
 
 import (
 	"context"
+	"crypto/tls"
 	"fmt"
 	"log"
+
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+
+	"github.com/pborman/uuid"
 )
 
 func main() {
+	ctx := context.Background()
 	log.Printf("Starting Teleport client...")
-	client, err := connectClient()
+
+	tlsConfig := &tls.Config{}
+	// Create valid tlsConfig here to load TLS credentials.
+
+	clt, err := auth.NewClient(client.Config{
+		// Addrs is the Auth Server address, must be local.
+		Addrs: []string{"localhost:3025"},
+		// Multiple credentials can be provided, and the first credentials to
+		// successfully authenticate an open connection to the client will be used.
+		Credentials: []client.Credentials{
+			client.LoadKeyPair(
+				"certs/access-admin.crt",
+				"certs/access-admin.key",
+				"certs/access-admin.cas",
+			),
+			client.LoadIdentityFile("certs/access-admin-identity"),
+			client.LoadTLS(tlsConfig),
+		},
+	})
 	if err != nil {
 		log.Fatalf("Failed to create client: %v", err)
 	}
-	defer client.Close()
+	defer clt.Close()
 
-	ctx := context.Background()
+	if err := demoClient(ctx, clt); err != nil {
+		log.Printf("error(s) in demoClient: %v", err)
+	}
+}
+
+func demoClient(ctx context.Context, clt *auth.Client) (err error) {
+	// Create a new access request for the `access-admin` user to use the `admin` role.
+	accessReq, err := types.NewAccessRequest(uuid.New(), "access-admin", "admin")
+	if err != nil {
+		return fmt.Errorf("failed to make new access request: %v", err)
+	}
+	if err = clt.CreateAccessRequest(ctx, accessReq); err != nil {
+		return fmt.Errorf("failed to create access request: %v", err)
+	}
+	log.Printf("Created access request: %v", accessReq)
 
-	fmt.Println("")
-	roleCRUD(ctx, client)
+	defer func() {
+		if err2 := clt.DeleteAccessRequest(ctx, accessReq.GetName()); err2 != nil {
+			err = fmt.Errorf("%v\nfailed to delete access request: %v", err, err2)
+			return
+		}
+		log.Println("Deleted access request")
+	}()
```

Comment:
this is a bit complicated for example code, requires some Go experience to understand the error handling
call `DeleteAccessRequest` at the end of the function, or just don't call it at all.